stdlib/public/core/StringCharacterView.swift

```diff
@@ -207,6 +207,21 @@ extension String: BidirectionalCollection {
 
     if i == endIndex { return 0 }
 
+    // Fast path ASCII grapheme breaking.
+    if _fastPath(_guts.isFastASCII) {
+      return _guts.withFastUTF8 {
+        let scalar1 = $0[_unchecked: i._encodedOffset]
+
+        if scalar1 == 0xD, i._encodedOffset &+ 1 < $0.count {
+          let scalar2 = $0[_unchecked: i._encodedOffset &+ 1]
+
+          return scalar2 == 0xA ? 2 : 1
+        }
+
+        return 1
+      }
+    }
+
```

Comment:
Instead, how about if the first byte is not `0x0D` and the next is `<0xC0` (I think, but double check), we return 1? That would speed up partially-ASCII string as well.